diff --git a/recipes/certificate_server.rb b/recipes/certificate_server.rb
index 53a32a5..671a926 100644
--- a/recipes/certificate_server.rb
+++ b/recipes/certificate_server.rb
@@ -7,14 +7,13 @@
 server_info = OpenShiftHelper::NodeHelper.new(node)
 is_certificate_server = server_info.on_certificate_server?
 is_master_server = server_info.on_master_server?
-docker_version = node['is_apaas_openshift_cookbook']['openshift_docker_image_version']
 
 if is_certificate_server || is_master_server
   if node['is_apaas_openshift_cookbook']['deploy_containerized']
 
-    docker_image node['is_apaas_openshift_cookbook']['openshift_docker_master_image'] do
-      tag docker_version
-      action :pull_if_missing
+    execute 'Pull CLI docker image' do
+      command "docker pull #{node['is_apaas_openshift_cookbook']['openshift_docker_master_image']}:#{node['is_apaas_openshift_cookbook']['openshift_docker_image_version']}"
+      not_if "docker images  | grep #{node['is_apaas_openshift_cookbook']['openshift_docker_master_image']}.*#{node['is_apaas_openshift_cookbook']['openshift_docker_image_version']}"
     end
 
     bash 'Add CLI to master(s)' do
diff --git a/recipes/etcd_cluster.rb b/recipes/etcd_cluster.rb
index 502be5e..f6ecb6b 100644
--- a/recipes/etcd_cluster.rb
+++ b/recipes/etcd_cluster.rb
@@ -147,8 +147,9 @@ if is_etcd_server
   end
 
   if node['is_apaas_openshift_cookbook']['deploy_containerized']
-    docker_image node['is_apaas_openshift_cookbook']['openshift_docker_etcd_image'] do
-      action :pull_if_missing
+    execute 'Pull ETCD docker image' do
+      command "docker pull #{node['cookbook-openshift3']['openshift_docker_etcd_image']}"
+      not_if "docker images  | grep #{node['cookbook-openshift3']['openshift_docker_etcd_image']}"
     end
 
     template "/etc/systemd/system/#{node['is_apaas_openshift_cookbook']['etcd_service_name']}.service" do
diff --git a/recipes/node.rb b/recipes/node.rb
index 72faa9b..260f024 100644
--- a/recipes/node.rb
+++ b/recipes/node.rb
@@ -8,7 +8,6 @@ server_info = OpenShiftHelper::NodeHelper.new(node)
 node_servers = server_info.node_servers
 certificate_server = server_info.certificate_server
 is_node_server = server_info.on_node_server?
-docker_version = node['is_apaas_openshift_cookbook']['openshift_docker_image_version']
 
 ose_major_version = node['is_apaas_openshift_cookbook']['deploy_containerized'] == true ? node['is_apaas_openshift_cookbook']['openshift_docker_image_version'] : node['is_apaas_openshift_cookbook']['ose_major_version']
 path_certificate = node['is_apaas_openshift_cookbook']['use_wildcard_nodes'] ? 'wildcard_nodes.tgz.enc' : "#{node['fqdn']}.tgz.enc"
@@ -45,14 +44,15 @@ if is_node_server
   end
 
   if node['is_apaas_openshift_cookbook']['deploy_containerized']
-    docker_image node['is_apaas_openshift_cookbook']['openshift_docker_node_image'] do
-      tag docker_version
-      action :pull_if_missing
+    execute 'Pull NODE docker image' do
+      command "docker pull #{node['is_apaas_openshift_cookbook']['openshift_docker_node_image']}:#{node['is_apaas_openshift_cookbook']['openshift_docker_image_v
+ersion']}"
+      not_if "docker images  | grep #{node['is_apaas_openshift_cookbook']['openshift_docker_node_image']}.*#{node['is_apaas_openshift_cookbook']['openshift_docker_image_version']}"
     end
 
-    docker_image node['is_apaas_openshift_cookbook']['openshift_docker_ovs_image'] do
-      tag docker_version
-      action :pull_if_missing
+    execute 'Pull OVS docker image' do
+      command "docker pull #{node['is_apaas_openshift_cookbook']['openshift_docker_ovs_image']}:#{node['is_apaas_openshift_cookbook']['openshift_docker_image_version']}"
+      not_if "docker images  | grep #{node['is_apaas_openshift_cookbook']['openshift_docker_ovs_image']}.*#{node['is_apaas_openshift_cookbook']['openshift_docker_image_version']}"
     end
 
     template '/etc/systemd/system/atomic-openshift-node-dep.service' do
